### PR TITLE
Add query by appId

### LINKED ISSUE #376: query type from SearchQuery interface from SDK is not correct
Currently `query`, from interface `SearchQuery` has type `string | string[] | number | number[]` however it should have the type `unknown` because the value can be an object with different structure. [Here](https://docs.nevermined.io/docs/tutorials/query-assets-catalog/) is possible to find several examples

### PR DESCRIPTION
**Pull request by r-marques**
## Description

This PR introduces an optional `appId` _SearchQuery_ parameter. When set the query will filter to return only ddos from that `appId`. The asset/nft `create` methods also have and new optional `appId` parameter that can be set when registering a DDO.

- simplified the query and search to use only one marketplace endpoint
- added appId as an optional SearchQuery parameter
- added query tests

## Is this PR related with an open issue?

Resolves https://github.com/nevermined-io/internal/issues/284
Resolves #376 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [x] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: src/nevermined/Assets.ts [hunk 1]
```diff
@@ -136,7 +136,8 @@ export class Assets extends Instantiable {
         nftAttributes: NFTAttributes | undefined,
         erc20TokenAddress: string | undefined,
         providers: string[] = [this.config.gatewayAddress],
-        txParams?: TxParameters
+        txParams?: TxParameters,
+        appId?: string
```
Comment (aaitor) [inline src/nevermined/Assets.ts:140]: maybe change order here so `txParams` is always the last parameter

File: src/nevermined/Assets.ts [hunk 2]
```diff
@@ -355,7 +356,8 @@ export class Assets extends Instantiable {
         encryptionMethod: EncryptionMethod,
         providers?: string[],
         nftMetadata?: string,
-        params?: TxParameters
+        params?: TxParameters,
```
Comment (aaitor) [inline src/nevermined/Assets.ts:359]: same

File: src/nevermined/Assets.ts [hunk 3]
```diff
@@ -470,7 +478,8 @@ export class Assets extends Instantiable {
         erc20TokenAddress?: string,
         preMint: boolean = true,
         nftMetadata?: string,
-        txParams?: TxParameters
+        txParams?: TxParameters,
```
Comment (aaitor) [inline src/nevermined/Assets.ts:481]: same

File: src/nevermined/Assets.ts [hunk 4]
```diff
@@ -518,7 +528,8 @@ export class Assets extends Instantiable {
         encryptionMethod: EncryptionMethod = 'PSK-RSA',
         providers?: string[],
         erc20TokenAddress?: string,
-        txParams?: TxParameters
+        txParams?: TxParameters,
```
Comment (aaitor) [inline src/nevermined/Assets.ts:531]: same

**Review by aaitor — APPROVED** [review]
lgtm

File: src/nevermined/Assets.ts (hunk 1 above)
Comment (r-marques) [reply src/nevermined/Assets.ts:140]: It would be better but the problem is that it introduces a breaking change

**Pull request merged**